src/main/java/io/mfj/expr/interop/PostgresAdapter.kt

```diff
@@ -0,0 +1,176 @@
+package io.mfj.expr.interop
+
+import io.mfj.expr.*
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.format.DateTimeFormatter
+
+object PostgresAdapter {
+  /**
+   * Serialize `trustedExpr` into a Postgres-compatible
+   * SQL expression string (e.g. for use in a WHERE clause)
+   *
+   * Input expression is NOT sanitized - do not use with
+   * expressions from untrusted sources, to avoid a SQL
+   * injection vector
+   */
+  fun toSqlExpression(trustedExpr: Expr): String {
+    return when (trustedExpr) {
+      is ExprConjunction -> toSql(trustedExpr)
+      is ExprLogicStatement -> toSql(trustedExpr)
+      else -> error("Unexpected type ${trustedExpr.javaClass}")
+    }
+  }
+
+  private fun toSql(conjunction: ExprConjunction): String {
+    return conjunction.params.joinToString(" ${conjunction.type.name} ") { expr ->
+      toSqlExpression(expr)
+    }.let { joined ->
+      if (conjunction.not) {
+        "NOT ($joined)"
+      } else {
+        if (conjunction.params.size > 1) {
+          "($joined)"
+        } else {
+          joined
+        }
+      }
+    }
+  }
+
+  private fun toSql(statement: ExprLogicStatement): String {
+    return when (statement.op) {
+      ExLogicOpType.REGEX_MATCH -> {
+        if (statement.left.getType() != ExDataType.STRING)
+          throw IllegalArgumentException( "Regex input must be a String, not ${statement.left.getType()}" )
+        val literal = (statement.right as? ExValueLit)?.value
+            ?: throw IllegalArgumentException("Regex pattern must be a simple literal")
+        val regex = if (literal is Regex) literal else Regex(literal.toString())
+        // the grammar doesn't seem to have any way to specify a case-insensitive regex
+        "${toSql(statement.left)} ~ '${regex.pattern}'"
+      }
+      ExLogicOpType.IN -> {
```

Comment:
Does Expr support substring checks with IN or CONTAINS?

Reply:
I will say, we don't have any of them in the measures currently (prosecutor or policing).  Maybe we should for SQL efficiency reasons - but, all current measures are like `(prosecuted=0 OR prosecuted=3 OR prosecuted=1)`

Reply:
@KarlBerger-MFJ it does not - looking at the [calculation implementation](https://github.com/measuresforjustice/expr/blob/794b0bae95f57b47526b20aca2288546a1119ab5/src/main/java/io/mfj/expr/ExprLogicStatement.kt#L160) for `IN`, the right operand must be a list, and it just checks whether anything in the list is equal to the left operand, so it's purely a straight mapping to SQL `IN`/`NOT IN`